Switches default ui to LINK instead of EMBED

WillyMcNamara-MA opened the pull request:
**Issue Ticket Number And Link**

*Please add a link to a corresponding Jira ticket if it exists.*

Link to Jira ticket: [[JIRA_BOARD-xxx](example.com)](https://moodys-ma-kys.atlassian.net/browse/OSCC-74)

**Description**

Makes the OSCC default to LINK instead of EMBED, since EMBED requires special permissions from PassFort.

**How Has This Been Tested?**

Locally for demo purposes

**Changes Required Elsewhere?**

N/A, this logic will change considerably with restructure. 

- [ ] Frontend
- [ x] Environment
- [ ] Other *(please specify below)*
- [ ] N/A

**Type Of Change**
- [ ] Bug Fix *(non-breaking change that fixes an issue)*
- [ ] New Feature *(non-breaking change that adds new functionality to the app)*
- [ ] Performance Improvement *(a change that enhances performance)*
- [ ] CI/CD *(any changes that impact the CI/CD pipeline)*
- [ ] Documentation
- [ ] Testing
- [ ] Maintenance/Infrastructure *(this may include things like logging, for example)*
- [ ] Security
- [ ] Refactoring *(updating the codebase to improve structure, but without changing core functionality)*
- [ ] Deprecations *(removing/phasing out parts of the codebase)*
- [ ] Style / Formatting *(formatting the code using a linter/formatter)*
- [ ] Other *(please specify below)*

WillyMcNamara-MA committed:
Removing prerequisite comment about Orbis access.

WillyMcNamara-MA committed:
Attempts to convert to LINK, running into error where passfort-integration.service methods are not accessible from passfort-integration.controller

WillyMcNamara-MA committed:
Working LINK in PassFort

WillyMcNamara-MA committed:
Merge branch 'main' into wm-dev-LINK-update

WillyMcNamara-MA committed:
Cleans up readme to reflect changes to LINK from EMBED

amckee250 commented on api/src/main.ts:
```diff
@@ -27,7 +27,7 @@ async function bootstrap() {
   );
   app.useGlobalFilters(
-    new FallbackExceptionFilter(),
+    //new FallbackExceptionFilter(),
```
Intentional?

amckee250 commented on api/src/main.ts:
```diff
@@ -27,7 +27,7 @@ async function bootstrap() {
   );
   app.useGlobalFilters(
-    new FallbackExceptionFilter(),
+    //new FallbackExceptionFilter(),
```
Intentional?

WillyMcNamara-MA replied:
Unintentional, removed.

amckee250 commented on api/static/demo_results/ALL_DATA.js:
```diff
@@ -4,10 +4,10 @@ export const ALL_DATA = {
      {
-      "type": "EMBED",
+      "type": "LINK", // was EMBED (need to write in from env ?)
       "url": "url_for_redirect/query_param_appended_here",
       "id": "8AA89547-89FC-4EAD-ACEC-FFA36F451337",
-      "label": "Example embed"
+      "label": "Example link" // was link (need to write in from env ?)
```
Intentional comment or todo item?

WillyMcNamara-MA replied:
this was a todo, just updated, thank you!

amckee250 reviewed (commented):
A couple possible adjustments for comments/commented out code.

To confirm, has this been tested via PF validation suite and profile check run? Have we confirmed the link is signed and ready for the FE to use to authenticate it was a valid PF generated link?

WillyMcNamara-MA committed:
Removes hard coded LINK and EMBED strings, pulling from .env
Removes comments
Comments back in filter which had been commented out in debugging.

WillyMcNamara-MA reviewed (commented):
Implemented these changes and pushing up now. Yes, this most recent push is tested in PassFort- swapping the .env variable between "EMBED" and "LINK", restarting the server and rerunning validation in PF results in a working version of either.

amckee250 commented on README.md:
```diff
@@ -38,7 +37,7 @@ This request is validated by your API through our VerifySignatureGuard before be
 
 PassFort expects a specific format for any check response, which is defined in [the documentation](https://passfort.github.io/integration-docs/?javascript#demo-checks). We've included a function createResponseObject in the api/src/utils/pf-integration.helpers.ts file which builds this formatted response. This response includes a 'result' property which let's PassFort know to display Pass/Fail/Par ...
 
-PassFort digests your API's response, displays the check results to the user, and calls your provided embed link to display your UI as an iFrame within PassFort. For security, we use a two step process here involving an HTML intermediary and the tokenization of your encoded endpoint/auth information. This lilypad gives us a safe way to finally display your UI within our PassFort environment. It's ...
```
Good catches all throughout here

amckee250 commented on api/static/demo_results/ALL_DATA.js:
```diff
@@ -4,10 +4,10 @@ export const ALL_DATA = {
   "external_resources": [
      {
-      "type": "EMBED",
+      "type": process.env.CHECK_TYPE || "LINK", // was EMBED (need to write in from env ?)
```
Comment still needed?

WillyMcNamara-MA replied:
hahaha 😭 . removed! thank you 🥇

amckee250 reviewed (approved):
Approved with a small question regarding if a comment is needed or not.

WillyMcNamara-MA committed:
Remove comment
